packages/eslint-plugin/src/util/explicitReturnTypeUtils.ts

```diff
@@ -256,10 +253,17 @@ function isValidFunctionExpressionReturnType(
   }
 
   // https://github.com/typescript-eslint/typescript-eslint/issues/653
-  return (
-    options.allowDirectConstAssertionInArrowFunctions === true &&
-    node.type === AST_NODE_TYPES.ArrowFunctionExpression &&
-    returnsConstAssertionDirectly(node)
+  if (
+    !options.allowDirectConstAssertionInArrowFunctions ||
+    node.type !== AST_NODE_TYPES.ArrowFunctionExpression
+  ) {
+    return false;
+  }
+
+  const { body } = node;
+
+  return isConstAssertion(
+    body.type === AST_NODE_TYPES.TSSatisfiesExpression ? body.expression : body,
   );
```

Comment:
You might wanna use a `while` loop here, like we do in other rules for recursive syntaxes.
```suggestion
  let body = node.body;
  while (body.type === AST_NODE_TYPES.TSSatisfiesExpression) {
    body = body.expression;
  }

  return isConstAssertion(body);
```
Not sure if there's any real world use case but here's a test case 🤷 
```ts
type FooBar = {
  foo: string,
  bar: number,
};

type EmptyObjectType = {}

export const b = () => ({
  foo: "FOOO",
  bar: 123,
} as const satisfies FooBar satisfies EmptyObjectType); 
```

Reply:
That's interesting, I didn't consider this edge case, thanks 👍

Reply:
I added the necessary tests to cover this.